src/ripple/basics/tagged_integer.h

```diff
@@ -157,6 +163,19 @@ class tagged_integer
         return tagged_integer (lhs.m_value - rhs);
     }
 
+    template <class OtherInt>
+    friend
+    typename std::enable_if <
+        std::is_integral <OtherInt>::value &&
+            sizeof (OtherInt) <= sizeof (Int),
```

Comment:
So the `ops` would be with respect to the an arbitrary integral type that played nicely with the internal type?

I'm also tempted to make "safe" overloads that define arithmetic within the tagged_type.  That is, require the client to convert to the tagged type, e.g.
```
using foo = tagged_integer<foo_tag,  uint32> 
foo a;
a = a % 10; // not-defined
a = a % foo{10};
```